pyvista/core/filters/image_data.py

```diff
@@ -1019,6 +1020,7 @@ def contour_labels(  # type: ignore[misc]
         select_outputs: int | VectorLike[int] | None = None,
         pad_background: bool = True,
         output_mesh_type: Literal['quads', 'triangles'] | None = None,
+        fast_mode: bool = False,
```

Comment:
This is a great idea, and I completely agree that it contributes to speeding things up. However, from a user's perspective, rather than emphasizing speed, it’s more beneficial for code clarity to clearly show what’s happening internally. Therefore, I would like to propose the variable name `remove_cell`.
```suggestion
        remove_cell: bool = False,
```

Reply:
What about `ignore_internal_boundary_effects` ?

It's more complicated than just removing cells. It's including internal cells and then removing them for the purpose of improving how the external cells are generated.

With the new option there are effectively two versions of `'external'` boundaries. So how can we easily differentiate the two? My concern with using language that refers to implementation details is that it might appear contradictory. E.g. doing 

`mesh.contour_labels('external', ignore_internal_boundary_effects=True)` 

May seem odd because both the term internal and external appear.

Reply:
What about removing the new `fast_mode` keyword and adding a new `'strict_external'` boundary style instead?

Reply:
+100 strict_external